tests/e2e/mandatory_modules_metrics_test.go

```diff
@@ -0,0 +1,103 @@
+package e2e_test
+
+import (
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var _ = Describe("Mandatory Module Metrics", Ordered, func() {
+	kyma := NewKymaWithSyncLabel("kyma-sample", "kcp-system", v1beta2.DefaultChannel,
+		v1beta2.SyncStrategyLocalSecret)
+
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
+
+	Context("Given SKR Cluster", func() {
+		const deployName = "template-operator-v1-controller-manager"
+		It("Then mandatory module is installed on the SKR cluster", func() {
+			Eventually(DeploymentIsReady).
+				WithContext(ctx).
+				WithArguments(runtimeClient, deployName,
+					TestModuleResourceNamespace).
+				Should(Succeed())
+			By("And the SKR Module Default CR is in a \"Ready\" State", func() {
+				Eventually(CheckSampleCRIsInState).
+					WithContext(ctx).
+					WithArguments("sample-yaml", "kyma-system", runtimeClient, "Ready").
+					Should(Succeed())
+			})
+			By("And the KCP Kyma CR is in a \"Ready\" State", func() {
+				Eventually(KymaIsInState).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+					Should(Succeed())
+			})
+
+			By("And count of Mandatory Module State Metric in \"Ready\" State is 1", func() {
+				Eventually(GetMandatoryModuleStateMetric).
+					WithContext(ctx).
+					WithArguments(kyma.GetName(), "template-operator", string(shared.StateReady)).
+					Should(Equal(1))
+			})
+
+			By("And count of Mandatory ModuleTemplates Metric is 1", func() {
+				Eventually(GetMandatoryModuleTemplateCountMetric).
+					WithContext(ctx).
+					Should(Equal(1))
+			})
+		})
+
+		It("When the mandatory ModuleTemplate is removed", func() {
+			Eventually(DeleteCR).
```

Comment:
There is the [DeleteModuleTemplate](https://github.com/kyma-project/lifecycle-manager/blob/165c9f578d846cf1a1aa48a8e9245e384f1c948c/pkg/testutils/moduletemplate.go#L74) method that you could use here instead of the DeleteCR

Reply:
This we actually cannot use, since this `DeleteModuleTemplate` func uses the `GetModuleTemplate` func given by our test utils. And this uses the `TemplateLookup` to fetch the ModuleTemplate - but only REGULAR templates and not mandatory ModuleTemplates. So, we could either introduce a `DeleteMandatoryModuleTemplate` function, or adapt the existing with a new bool input parameter, to switch between different LookUp mechanisms (in my opinion to overengineered) or we just delete the CR, since we know what the name is.